core/src/main/java/org/elasticsearch/env/NodeEnvironment.java

```diff
@@ -1003,8 +1003,12 @@ private static void tryWriteTempFile(Path path) throws IOException {
         if (Files.exists(path)) {
             Path resolve = path.resolve(".es_temp_file");
             try {
-                Files.createFile(resolve);
-                Files.deleteIfExists(resolve);
```

Comment:
I'd prefer to delete the temp file first as well but it looks like to be consistent with #19036 we should do the delete in a finally block. That way if the file exists we'll nuke it in the finally block as we abort startup. I guess this has the advantage of failing once for the user so they know it is a problem.... I'm not sold on it.

The bottom line is that what we have now is wrong and if you move the delete to a `finally` block I can merge this without further debate. We can then open an issue or another PR to discuss deleting the file first. That way we have *a* fix in and we can more leisurely decide what the right solution is.